pkg/scheduler/core/pod_scheduler/pod_scheduler_test.go

```diff
@@ -84,21 +89,46 @@ func TestSelectHost(t *testing.T) {
 			},
 			possibleHosts: sets.NewString("machine1.1", "machine1.2", "machine1.3"),
 			expectsErr:    false,
+			podOwner:      "Alice",
 		},
 		{
 			name:          "empty priority list",
 			list:          []framework.NodeScore{},
 			possibleHosts: sets.NewString(),
 			expectsErr:    true,
+			podOwner:      "Bob",
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			lastCacheSize := 0
 			// increase the randomness
 			for i := 0; i < 10; i++ {
 				// TODO: add unit test cases for caching node logic
-				got, err := scheduler.selectHostAndCacheResults(test.list, &v1.Pod{}, "", "", &framework.UnitSchedulingRequest{})
+				got, err := scheduler.selectHostAndCacheResults(test.list, &v1.Pod{}, test.podOwner, "", &framework.UnitSchedulingRequest{})
+				nodes := scheduler.isolatedCache.GetOrderedNodesForPodOwner(test.podOwner)
+				currCacheSize := len(nodes)
+				expectedCacheSize := lastCacheSize
+				// exclude the selected node
+				if len(test.list) > 1 {
+					expectedCacheSize += len(test.list) - 1
+				}
+
+				if currCacheSize != expectedCacheSize {
+					t.Errorf("Unexpected cache size, expected %d but got %d", currCacheSize, expectedCacheSize)
+				}
+				lastCacheSize = currCacheSize
+
+				// Check newly cached nodes
+				newlyCachedNodes := nodes[lastCacheSize:]
+				selectedNode := got
```

Comment:
why it?

Reply:
can we use `selectedNode` directly?

Reply:
I use slice here because the cache will not clear during the testing loop, and the cached node will be accumulated.

Reply:
i mean why use `selectedNode := got`

Reply:
sure, we can use selectedNode directly. Some following codes use the 'got' var, so I don't want to affect them at the beginning. I directly use var "selectedNode" in the new commit.